share/ruby-build/3.2.0-dev

```diff
@@ -1,2 +1,2 @@
-install_package "openssl-3.0.2" "https://www.openssl.org/source/openssl-3.0.2.tar.gz#98e91ccead4d4756ae3c9cde5e09191a8e586d9f4d50838e7ec09d6411dfdb63" mac_openssl --if has_broken_mac_openssl
+install_package "openssl-3.0.2" "https://www.openssl.org/source/openssl-3.0.2.tar.gz#98e91ccead4d4756ae3c9cde5e09191a8e586d9f4d50838e7ec09d6411dfdb63" openssl --if needs_openssl
```

Comment:
Ruby 3.2 seems to want OpenSSL 3, and `needs_openssl` will evaluate to true if system OpenSSL is v3. Is this desired? This will cause an additional OpenSSL to be compiled even though system OpenSSL is likely sufficient